src/ThingsBoard.h

```diff
@@ -10,6 +10,8 @@
 // Library includes.
 #include <stdarg.h>
 #include <TBPubSubClient.h>
+#include <assert.h>
```

Comment:
The include position is good and seems fine, but it only needs to be included / can most of the time be included if C++ STL is supported therefore a `#ifdef` around the `#include` needs to be added.

```c++
#if THINGSBOARD_ENABLE_STL
#include <assert.h>
#endif // THINGSBOARD_ENABLE_STL
````

Reply:
tried it and got the mentioned error again.

Reply:
That include now has to be moved below `Configuration.h`, because in that file the `THINGSBOARD_ENABLE_STL` is defined.
Additionally the extra spaces are not needed, 2 spaces to private and then another 2 is done in other code files in the library as well.

Would be nice if you could revert the latest commit and then add this code below the local includes.

```c++
#if THINGSBOARD_ENABLE_STL
#include <assert.h>
#endif // THINGSBOARD_ENABLE_STL
```

Reply:
Sure. Will get to it this afternoon.